### PR TITLE
Update HasGroups spec

### PR DESCRIPTION
**Pull request by niklassiemer**

### CODE REVIEW

File: specs/has_groups.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ also provide a way to actually access the contained objects.  This is done via `
 distinguished from a node by containing other objects it also represents a vertex "containing" or "linked-to" other
 vertices.  From this we formulate the requirements for this interface
 
-1. `__getitem__()` must return a non-`None` object for each string returned by `list_nodes()` and `list_groups()`.
+1. `__getitem__()` must return any or a non-`None` object for each string returned by `list_nodes()` or `list_groups()`, repectively.
```
Comment (pmrv) [inline specs/has_groups.md]: > must return any or a non-`None` object

Not sure what that means.

Comment (liamhuber) [reply specs/has_groups.md]: "any of"?

Comment (niklassiemer) [reply specs/has_groups.md]: Sorry,
'any object or a non-None type object' since getitem of a node might return `None`...

Comment (niklassiemer) [reply specs/has_groups.md]: Is it better now?

**Comment by niklassiemer** [comment]
I just realized we are violating our spec! `Project.list_nodes()` returns a list of jobs. However, any job isinstance of `HasGroups`!

**Pull request closed without merging**